src/main/java/persistence/sql/dml/impl/DefaultEntityManager.java

```diff
@@ -1,45 +1,49 @@
 package persistence.sql.dml.impl;
 
 import jakarta.persistence.EntityExistsException;
+import persistence.sql.EntityLoaderFactory;
 import persistence.sql.clause.Clause;
 import persistence.sql.context.EntityPersister;
 import persistence.sql.context.PersistenceContext;
 import persistence.sql.dml.EntityManager;
 import persistence.sql.dml.MetadataLoader;
+import persistence.sql.loader.EntityLoader;
 
 import java.lang.reflect.Field;
 import java.util.List;
-import java.util.logging.Logger;
 
 public class DefaultEntityManager implements EntityManager {
-    private static final Logger logger = Logger.getLogger(DefaultEntityManager.class.getName());
-
     private final PersistenceContext persistenceContext;
     private final EntityPersister entityPersister;
+    private final EntityLoaderFactory entityLoaderFactory;
 
     public DefaultEntityManager(PersistenceContext persistenceContext, EntityPersister entityPersister) {
         this.persistenceContext = persistenceContext;
         this.entityPersister = entityPersister;
+        this.entityLoaderFactory = EntityLoaderFactory.getInstance();
     }
 
     @Override
-    public <T> void persist(T entity) {
+    public <T> T persist(T entity) {
         if (entity == null) {
             throw new IllegalArgumentException("Entity must not be null");
         }
-        MetadataLoader<?> loader = new SimpleMetadataLoader<>(entity.getClass());
 
-        if (!isNew(entity, loader)) {
+        if (!isNew(entity)) {
             throw new EntityExistsException("Entity already exists");
         }
 
-        Object id = entityPersister.insert(entity, loader);
-        updatePrimaryKeyValue(entity, id, loader);
+        Object id = entityPersister.insert(entity);
         persistenceContext.add(id, entity);
+
+        return entity;
     }
 
-    private <T> boolean isNew(Object entity, MetadataLoader<T> loader) {
+    private <T> boolean isNew(Object entity) {
```

Comment:
```suggestion
    private boolean isNew(Object entity) {
```

더 이상 제네릭이 필요 없어보이네요 :)